include/assembler.cpp

```diff
@@ -53,10 +54,21 @@ uint16_t swap_bytes(uint16_t number) {
     return result;
 }
 
-void assembleLine(std::string line, int lineNum, std::string romName) {
+int assembleLine(const std::string& line, int lineNum, const std::string& romName) {
     std::string instruction;
 
-    instruction = line.substr(0, 4);
+    std::stringstream noNewlineStream;
+    // Operator << discards newline
+    noNewlineStream << line;
+
+    std::string noNewlineLine = noNewlineStream.str();
+
+    if (noNewlineLine.empty()) {
+        std::cout << lineNum << " is an empty line, skipping..." << std::endl;
+        return 0;
+    }
+
+    instruction = noNewlineLine.substr(0, 4);
```

Comment:
This assumes all mnemonics are 4 characters long. Maybe use a `#define MNEMONIC_LENGTH 4`